### ISSUE TITLE
New Cache system for SuiteCRM

### ISSUE THREAD

**Issue by Abuelodelanada** [salesagility/SuiteCRM#3005]
#### Expected Behavior

A new caching method could be implemented using the CacheLite library[1] to prevent many queries being executed to the database, and as a new cache method for SuiteCRM.

[1] https://pear.php.net/package/Cache_Lite/

#### Actual Behavior

As we know SuiteCRM has many cache classes but use by default SugarCacheAbstract class. This cache method use RAM. This is good because RAM is fast, but the cache lasts for a limited time: the request, so in every request has to regenerate the cache. Besides the cache is not shared between users.

CacheLite stores the cache in files, which is slower than using RAM, but has many other benefices:

- The cache last much more time than the request, and you can configure that duration ($sugar_config['CacheLite']['LifeTime'] = 600; in config_override.php)
- The cache is shared between users.
- Reseting the cache is as simple as deleting the cache files.
- You can separate the cache in 'groups', so you can delete the cache of certain groups and the whole.

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
